rbi/core/enumerator.rbi

```diff
@@ -154,6 +154,18 @@ class Enumerator < Object
   sig { params(initial: T.untyped, block: T.proc.params(arg: T.untyped).void).returns(T::Enumerator[T.untyped]) }
   def self.produce(initial = nil, &block); end
 
+  # Returns an enumerator object generated from this enumerator and a given 
+  # enumerable.
+  #
+  # ### Examples
+  #
+  # ```ruby
+  # e = (1..3).each + [4, 5]
+  # e.to_a #=> [1, 2, 3, 4, 5]
+  # ```
+  sig { params(enum: T::Enumerable[T.untyped]).returns(T::Enumerator::Chain[T.untyped]) }
```

Comment:
👋 Why does this not use generics to avoid depending on `T.untyped`?

Reply:
Generics really throw me every time I have to work with them but from how I understand it the resulting chain would look something like

```suggestion
  sig { type_parameters(:T).params(enum: T::Enumerable[type_parameter(:T)]).returns(T::Enumerator::Chain[T.any(type_parameter(:T), self::Elem) }
```

What would be the correct way to write this?

Reply:
yep, I believe so

Reply:
You can omit the `self::`, just `Elem` is fine.

Reply:
done! https://github.com/sorbet/sorbet/pull/8226/commits/5150cd820814775b0bbde29dec730a447cfbb2bd